### ISSUE TITLE
Adding new integration hangs forever, when home assistant is not reloaded

### ISSUE THREAD

**Issue by tiramiseb** [Hyundai-Kia-Connect/kia_uvo#603]
**Region and Brand of car**

Europe, Hyundai Ioniq 16/19

**Version of the integration**

v2.6.0

**Describe the bug**

I use HA in a container. Right after starting the container (with kia_uvo correctly installed in custom_components), when I try to add an integration with kia_uvo, the wait screen (before displaying the login form) hangs there forever (or I get a "gateway timeout" when behind my nginx server).

After restarting homeassistant from the "development tools" page, it works as expected.

**Debug logs if an error occurred**

Absolutely no log from kia_uvo when there is a timeout. I have even added `_LOGGER.error("foobar")` in `__init__.py` ("root" context), and "foobar" does not display.

Once homeassistant is restarted from its UI, I have "foobar" in the logs when the component is loaded and everything works fine.

**To Reproduce**

See the bug description, I don't know what to add.

**Expected behavior**

No hanging, and component initializes correctly without restarting home assistant manually.

**Screenshots**

Nothing interesting.

**Additional context**

It looks like HA installs custom components dependencies (in this case, hyundai_kia_connect_api) only when an integration is added. And it feels like it works only if the dependency is installed before the homeassistant process starts. But I may be totally wrong in my analysis...

Please note currently I have no "force update" working, including with the Android app. Either there is a problem with Hyundai infrastructure, or it may be because the car is at the repair shop... Therefore, after restarting homeassistant again, it takes forever to load and I end up with a "Request timeout" error in the log, but I think it is not related to the bug I'm reporting.

**Comment by cdnninja** [comment]
No force update could be related.  The system doesn't like it when the car is down for a long time.   Have you enabled debug logging?

**Comment by tiramiseb** [comment]
Yep, I have even come to enable debug logging for everything, and then hiding too verbose and unrelated stuff one by one (see below), in order to have as much logs as possible.

Nothing relevant, no debug log from kia_evo, no debug log related to kia_evo from core, etc.

```plain
      logger:
        default: debug
        logs:
          zeroconf: info
          aiowebostv: info
          homeassistant.components.ssdp: info
          homeassistant.components.zeroconf: info
          async_upnp_client.traffic.ssdp: info
          async_upnp_client.traffic.upnp: info
          async_upnp_client.search: info
          async_upnp_client.server: info
          async_upnp_client.advertisement: info
          homeassistant.components.nextdns: info
          nextdns: info
          homeassistant.components.upnp: info
          async_upnp_client.ssdp: info
          pyipp.parser: info
          homeassistant.components.mobile_app.webhook: info
          homeassistant.components.http.view: info
          homeassistant.components.dlna_dms: info
          paho.mqtt.client: info
          homeassistant.components.ipp.coordinator: info
          async_upnp_client.traffic.upnp: info
          bluetooth_adapters.dbus: info
          async_upnp_client.ssdp_listener: info
          homeassistant.components.bluetooth: info
```

**Comment by cdnninja** [comment]
Anything from the api module? It won't come from kia_uvo.

**Comment by tiramiseb** [comment]
I haven't seen anything relevant, no. Or maybe there was something else I needed to change in order to have the api module show debug logs?

**Comment by tiramiseb** [comment]
Okay, I think I found the problem. It occured too on a component I'm currently developing.

I have NginX as a reverse proxy in front of Home Assistant. My server is pretty slow (Atom 230, I have this computer since 2009).

New python packages (ie. components requirements) installation take a long time. Default NginX configuration has a 1 minute timeout on proxy responses, then NginX kills the connection.

It looks like Home Assistant does not like it, because even when accessing HA directly I could not add the integration without restarting HA.

I have increased the proxy timeout to 5 minutes, to leave enough time to install the requirements, and it seems to work.
